servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/DefaultLoadBalancer.java

```diff
@@ -508,7 +508,9 @@ private void sequentialUpdateUsedHosts(List<PrioritizedHostImpl<ResolvedAddress,
         for (PrioritizedHostImpl<?, ?> host : nextHosts) {
             host.loadBalancingWeight(host.serviceDiscoveryWeight());
         }
-        this.hostSelector = hostSelector.rebuildWithHosts(priorityStrategy.prioritize(usedHosts));
+        nextHosts = priorityStrategy.prioritize(nextHosts);
+        this.hostSelector = hostSelector.rebuildWithHosts(nextHosts);
+        loadBalancerObserver.onHostSetChanged(Collections.unmodifiableCollection(nextHosts));
```

Comment:
After the 2nd look: `priorityStrategy` returns a `List`. Why does the observer provide a `Collection`? Just to align with SD discovery publisher contract? Will `List` be beneficial to signal Host priorities to users of LB-observer?

Reply:
We talked offline and the static type of `Collection` is fine and I've used the `unmodifiableList` method to make the dynamic type more helpful.